forge/test/mlir/llama/tests/test_specific_ops_llama32.py

```diff
@@ -0,0 +1,480 @@
+
+
+@pytest.mark.parametrize(
+    "shapes",
+    [
+        (1, 11, 2048),
+    ],
+)
+@pytest.mark.xfail(reason="pcc < 0.75")
+def test_reduce_avg(shapes):
+    class ReduceAvg(nn.Module):
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, x):
+            return torch.mean(x, dim=-1, keepdim=True)
+
+    inputs = [torch.rand(shapes)]
+    framework_model = ReduceAvg()
+    fw_out = framework_model(*inputs)
+
+    compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+    co_out = compiled_model(*inputs)[0].to("cpu")
+
+    assert compare_with_golden_pcc(fw_out, co_out, pcc=0.95)
+
+
+@pytest.mark.parametrize(
+    "shapes",
+    [
+        (1, 11, 8192),
+    ],
+)
+def test_sigmoid(shapes):
+    class Sigmoid(nn.Module):
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, x):
+            return torch.sigmoid(x)
+
+    inputs = [torch.rand(shapes)]
+    framework_model = Sigmoid()
+    fw_out = framework_model(*inputs)
+
+    compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+    co_out = compiled_model(*inputs)[0].to("cpu")
+
+    assert compare_with_golden_pcc(fw_out, co_out, pcc=0.99)
+
+
+@pytest.mark.parametrize(
+    "shapes",
+    [
+        ((1, 11, 1),),
+    ],
+)
+def test_reciprocal(shapes):
+    class Reciprocal(nn.Module):
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, x):
+            return torch.reciprocal(x)
+
+    inputs = [torch.rand(shapes[0]) + 0.1]
+    framework_model = Reciprocal()
+    fw_out = framework_model(*inputs)
+
+    compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+    co_out = compiled_model(*inputs)[0].to("cpu")
+
+    assert compare_with_golden_pcc(fw_out, co_out, pcc=0.99)
+
+
+@pytest.mark.parametrize(
+    "source_and_target_shape",
+    [
+        ((1, 11, 2048), (11, 2048)),
+        ((1, 11, 2048), (1, 11, 32, 64)),
+        ((1, 11, 2048), (1, 11, 2048)),
+        ((1, 32, 11, 64), (32, 11, 64)),
+        ((1, 32, 11, 64), (1, 32, 11, 64)),
+        ((11, 512), (1, 11, 8, 64)),
+        ((1, 8, 4, 11, 64), (32, 11, 64)),
+        ((1, 8, 4, 11, 64), (1, 32, 11, 64)),
+        ((32, 11, 11), (1, 32, 11, 11)),
+        ((32, 11, 11), (32, 11, 11)),
+        ((1, 32, 64, 11), (32, 64, 11)),
+        ((1, 11, 32, 64), (11, 2048)),
+        ((11, 8192), (1, 11, 8192)),
+    ],
+)
+def test_reshape(source_and_target_shape):
+    source_shape, target_shape = source_and_target_shape
+
+    if len(source_shape) != 4 or len(target_shape) != 4:
+        pytest.xfail("Reshape for dim != 4: Unhandled attribute type")
+
+    if source_shape != target_shape or source_and_target_shape == ((1, 32, 11, 64), (1, 32, 11, 64)):
+        pytest.xfail(
+            "Unable to reshape a tensor in TILE_LAYOUT to non-tile height and width! Please convert the tensor to ROW_MAJOR_LAYOUT first."
```

Comment:
Let's check on latest main if this one is fixed as well

Reply:
Same as embedding it seems